src/common.ts

```diff
@@ -98,13 +104,9 @@ export interface ReadContract {
 }
 
 export interface WriteContract {
-  writeInteraction<Input>({
-    functionName,
-    inputs,
-    evaluationOptions,
-  }: EvaluationParameters<
-    WriteParameters<Input>
-  >): Promise<WriteInteractionResult>;
+  writeInteraction<Input>(
+    ...[{ functionName, inputs }, options]: WriteArguments<Input>
```

Comment:
using a spread operator here is something i haven't seen before. can you confirm this is equivalent to `writeInteraction<Input>({ functionName, inputs}, options)`?

Reply:
eh kinda, it actually looks like `0_0_WriteParameters<Input>, 0_1_WriteOptions` - it is technically the correct type and is human readable, just a bit ugly. I mainly did it this way instead of `(params: WriteArguments<Input>)` to enable verbosity for the positional parameters in the type def.

I tried it without the spread operator as well but typescript then considers it a positional argument at the zeroth index.

Reply:
hm, i think i'd prefer `writeInteraction<Input>({ functionName, inputs}, { tags })`